### PR TITLE
BCP-47 support in Developer for Package Editor

### PR DESCRIPTION
**Pull request by mcdurdin**
(Note: renamed branch from #466)

This is needed as a prerequisite for the package editor to correctly support the package format (which uses BCP-47 codes in the languages section).

As a bonus it takes us a long way towards transitioning Keyman Developer away from LCID and EthnologueCodes and into BCP-47 nirvana, when we come do that feature.

This PR includes:

Standards data files in the top-level resources/standards-data folder.
A new project to generate Delphi source files from the standards-data, buildtools/build_standards_data.
A new class, TKMXFileLanguages which provides mappings from legacy KMX language codes to BCP-47 codes
A hacked up locale enumerator project that was used to generate lcid-to-bcp47 static data. This project is kept for reference but probably won't ever be required again.
A unit test for TKMXFileLanguages
Removal of old versions of standards-data-based source files, and some tidy-up.

### CODE REVIEW

File: windows/src/buildtools/build_standards_data/Keyman.System.BuildLCIDToBCP47Registry.pas [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+  try
+    LoadFromFile(MapFile);
+
+    FResult := TStringList.Create;
+    try
+      FResult.Add('unit Keyman.System.Standards.LCIDToBCP47Registry;');
+      FResult.Add('');
+      FResult.Add('interface');
+      FResult.Add('');
+      FResult.Add('// File-Date: '+FormatDateTime('yyyy-mm-dd hh:nn:ss', Now));
```
Comment (darcywong00) [inline windows/src/buildtools/build_standards_data/Keyman.System.BuildLCIDToBCP47Registry.pas]: Would it be good to format these in UTC?

Comment (mcdurdin) [reply windows/src/buildtools/build_standards_data/Keyman.System.BuildLCIDToBCP47Registry.pas]: It's really not important. It's generated every build. So, nope.

File: windows/src/buildtools/build_standards_data/Keyman.System.BuildISO6393Registry.pas [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+  s6391: string;
+
+  function GetField(var s: string): string;
+  var
+    i: Integer;
+  begin
+    i := 1;
+    while i <= Length(s) do
+    begin
+      if s[i] = #9 then
```
Comment (darcywong00) [inline windows/src/buildtools/build_standards_data/Keyman.System.BuildISO6393Registry.pas]: Can `#9` be changed to a constant?

Comment (mcdurdin) [reply windows/src/buildtools/build_standards_data/Keyman.System.BuildISO6393Registry.pas]: I probably should have but next time...

**Pull request merged**